browser/ui/views/split_view/split_view.h

```diff
@@ -0,0 +1,143 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_SPLIT_VIEW_SPLIT_VIEW_H_
+#define BRAVE_BROWSER_UI_VIEWS_SPLIT_VIEW_SPLIT_VIEW_H_
+
+#include <memory>
+#include <vector>
+
+#include "base/functional/callback_forward.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/raw_ref.h"
+#include "base/memory/weak_ptr.h"
+#include "base/types/pass_key.h"
+#include "brave/browser/ui/tabs/split_view_browser_data.h"
+#include "brave/browser/ui/tabs/split_view_browser_data_observer.h"
+#include "brave/browser/ui/views/split_view/split_view_layout_manager.h"
+#include "chrome/browser/ui/tabs/tab_model.h"
+#include "ui/base/metadata/metadata_header_macros.h"
+#include "ui/gfx/geometry/rounded_corners_f.h"
+#include "ui/views/view.h"
+
+namespace content {
+class WebContents;
+}  // namespace content
+
+namespace views {
+class WebView;
+}  // namespace views
+
+class BraveBrowserView;
+class Browser;
+class ContentsWebView;
+class DevToolsContentsResizingStrategy;
+class SplitViewLayoutManager;
+class SplitViewLocationBar;
+class SplitViewSeparator;
+
+// Contains a pair of contents container view.
+class SplitView : public views::View, public SplitViewBrowserDataObserver {
+  METADATA_HEADER(SplitView, views::View)
+ public:
+  using BrowserViewKey = base::PassKey<BraveBrowserView>;
+
+  SplitView(Browser& browser,
+            views::View* contents_container,
+            ContentsWebView* contents_web_view);
+
+  ~SplitView() override;
+
+  // This must be called before BrowserView sets the active web contents to
+  // contents web view, we don't observe tab strip model directly. This returns
+  // callback that should be called after swapping active web contents
+  using AfterSetWebContents =
```

Comment:
why does this one use a callback, but below we call `DidUpdateDevToolsForActiveContents`?

Reply:
We can avoid callback. Deleted and used same pattern(`Will`/`Did`).